File: Solutions/322. Coin Change/322. Coin Change.java
class Solution {
    public int coinChange(int[] coins, int amount) {
        int[] dp=new int[amount+1];
        Arrays.fill(dp,amount+9487);
        dp[0]=0;
        for(int i=1;i<=amount;++i)
            for(var c:coins)
                if(i>=c)
                    dp[i]=Math.min(dp[i],1+dp[i-c]);
        return dp[amount]==amount+9487?-1:dp[amount];
    }
}

File: Solutions/322. Coin Change/322. Coin Change.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp(amount+1,amount+9487);
        dp[0]=0;
        for(int i=1;i<=amount;++i)
            for(const auto& c:coins)
                if(i>=c)
                    dp[i]=min(dp[i],1+dp[i-c]);
        return dp[amount]==amount+9487?-1:dp[amount];
    }
};